apps/gitness/src/pages/repo/repo-commits.tsx

```diff
@@ -1,33 +1,54 @@
 import { ListActions, Spacer, Text } from '@harnessio/canary'
 import { BranchSelector, NoData, PaddingListLayout, PullRequestCommits, SkeletonList } from '@harnessio/playground'
 import { useGetRepoRef } from '../../framework/hooks/useGetRepoPath'
-import { useListBranchesQuery, useListCommitsQuery } from '@harnessio/code-service-client'
+import {
+  TypesCommit,
+  useFindRepositoryQuery,
+  useListBranchesQuery,
+  useListCommitsQuery
+} from '@harnessio/code-service-client'
+import { useEffect, useState } from 'react'
+import { normalizeGitRef } from '../../utils/git-utils'
 
 const filterOptions = [{ name: 'Filter option 1' }, { name: 'Filter option 2' }, { name: 'Filter option 3' }]
 const sortOptions = [{ name: 'Sort option 1' }, { name: 'Sort option 2' }, { name: 'Sort option 3' }]
 
 export default function RepoCommitsPage() {
   const repoRef = useGetRepoRef()
 
-  const { data: commitData, isFetching: isFetchingCommits } = useListCommitsQuery({
+  const { data: repository } = useFindRepositoryQuery({ repo_ref: repoRef })
+
+  const { data: branches, isFetching: isFetchingBranches } = useListBranchesQuery({
     repo_ref: repoRef,
     queryParams: { page: 0, limit: 10 }
   })
-  const { data: branches, isFetching: isFetchingBranches } = useListBranchesQuery({
+  const [selectedBranch, setSelectedBranch] = useState<string>('')
+
+  const { data: commitData, isFetching: isFetchingCommits } = useListCommitsQuery({
     repo_ref: repoRef,
-    queryParams: { page: 0, limit: 10 }
+    queryParams: { page: 0, limit: 10, git_ref: normalizeGitRef(selectedBranch) }
   })
 
+  useEffect(() => {
+    if (repository) {
+      setSelectedBranch(repository?.default_branch || '')
+    }
+  }, [repository])
+
+  const selectBranch = (branch: string) => {
+    setSelectedBranch(branch)
+  }
+
   const renderContent = () => {
     if (isFetchingCommits) {
       return <SkeletonList />
     }
-    // @ts-expect-error remove "@ts-expect-error" once type issue for "content" is resolved
-    if (commitData?.content?.commits?.length) {
+    // @ts-expect-error remove "@ts-expect-error" once CodeServiceClient Response for useListCommitsQuery is fixed
+    if (commitData?.commits?.length) {
       return (
         <PullRequestCommits
-          // @ts-expect-error remove "@ts-expect-error" once type issue for "content" is resolved
-          data={commitData?.content.commits.map((item: TypesCommit) => ({
+          // @ts-expect-error remove "@ts-expect-error" once CodeServiceClient Response for useListCommitsQuery is fixed
+          data={commitData?.commits.map((item: TypesCommit) => ({
```

Comment:
Is commits query the only one like this ? @vardanbansal-harness

Reply:
Yeah I saw that, by definition `useListCommitsQuery` should return a response of type `TypesListCommitResponse[]` but it's only returning one object of type `TypesListCommitResponse` instead.

This issue is on the similar lines I've been seeing for repository listing as well.

Reply:
@shaurya-harness I don't see `commits` being refetched on `branch` selection.

Reply:
@vardanbansal-harness  that is taken care of as the state var selectedBranch is passed in queryParams